from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(1000, 500)
        self.centralWidget = QtWidgets.QWidget(MainWindow)
        self.centralWidget.setObjectName("centralWidget")
        self.horizontalLayout = QtWidgets.QHBoxLayout(self.centralWidget)
        self.horizontalLayout.setContentsMargins(11, 11, 11, 11)
        self.horizontalLayout.setSpacing(6)
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.widget = QtWidgets.QWidget(self.centralWidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Maximum, QtWidgets.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.widget.sizePolicy().hasHeightForWidth())
        self.widget.setSizePolicy(sizePolicy)
        self.widget.setObjectName("widget")
        self.LeftOperation = QtWidgets.QVBoxLayout(self.widget)
        self.LeftOperation.setContentsMargins(11, 11, 11, 11)
        self.LeftOperation.setSpacing(6)
        self.LeftOperation.setObjectName("LeftOperation")
        self.gp_file_option = QtWidgets.QGroupBox(self.widget)
        self.gp_file_option.setObjectName("gp_file_option")
        self.verticalLayout = QtWidgets.QVBoxLayout(self.gp_file_option)
        self.verticalLayout.setContentsMargins(11, 11, 11, 11)
        self.verticalLayout.setSpacing(6)
        self.verticalLayout.setObjectName("verticalLayout")
        self.btn_openImg = QtWidgets.QPushButton(self.gp_file_option)
        self.btn_openImg.setObjectName("btn_openImg")
        self.verticalLayout.addWidget(self.btn_openImg)
        self.LeftOperation.addWidget(self.gp_file_option)
        self.gp_ability_option = QtWidgets.QGroupBox(self.widget)
        self.gp_ability_option.setObjectName("gp_ability_option")
        self.gridLayout_4 = QtWidgets.QGridLayout(self.gp_ability_option)
        self.gridLayout_4.setContentsMargins(11, 11, 11, 11)
        self.gridLayout_4.setSpacing(6)
        self.gridLayout_4.setObjectName("gridLayout_4")
        self.ck_cut = QtWidgets.QCheckBox(self.gp_ability_option)
        self.ck_cut.setChecked(True)
        self.ck_cut.setObjectName("ck_cut")
        self.gridLayout_4.addWidget(self.ck_cut, 1, 0, 1, 1)
        self.ck_noline = QtWidgets.QCheckBox(self.gp_ability_option)
        self.ck_noline.setLayoutDirection(QtCore.Qt.LeftToRight)
        self.ck_noline.setObjectName("ck_noline")
        self.gridLayout_4.addWidget(self.ck_noline, 0, 0, 1, 1)
        self.ck_print = QtWidgets.QCheckBox(self.gp_ability_option)
        self.ck_print.setObjectName("ck_print")
        self.gridLayout_4.addWidget(self.ck_print, 0, 1, 1, 1)
        self.ck_join = QtWidgets.QCheckBox(self.gp_ability_option)
        self.ck_join.setChecked(True)
        self.ck_join.setObjectName("ck_join")
        self.gridLayout_4.addWidget(self.ck_join, 1, 1, 1, 1)
        self.LeftOperation.addWidget(self.gp_ability_option)
        self.gp_threshold = QtWidgets.QGroupBox(self.widget)
        self.gp_threshold.setObjectName("gp_threshold")
        self.gridLayout_2 = QtWidgets.QGridLayout(self.gp_threshold)
        self.gridLayout_2.setContentsMargins(11, 11, 11, 11)
        self.gridLayout_2.setSpacing(6)
        self.gridLayout_2.setObjectName("gridLayout_2")
        self.sb_ratio_value = QtWidgets.QLabel(self.gp_threshold)
        self.sb_ratio_value.setMinimumSize(QtCore.QSize(30, 0))
        self.sb_ratio_value.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.sb_ratio_value.setObjectName("sb_ratio_value")
        self.gridLayout_2.addWidget(self.sb_ratio_value, 6, 2, 1, 1)
        self.sb_ratio = QtWidgets.QSlider(self.gp_threshold)
        self.sb_ratio.setEnabled(False)
        self.sb_ratio.setMaximum(100)
        self.sb_ratio.setProperty("value", 30)
        self.sb_ratio.setOrientation(QtCore.Qt.Horizontal)
        self.sb_ratio.setObjectName("sb_ratio")
        self.gridLayout_2.addWidget(self.sb_ratio, 6, 1, 1, 1)
        self.sb_binary_value = QtWidgets.QLabel(self.gp_threshold)
        self.sb_binary_value.setMinimumSize(QtCore.QSize(30, 0))
        self.sb_binary_value.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.sb_binary_value.setObjectName("sb_binary_value")
        self.gridLayout_2.addWidget(self.sb_binary_value, 4, 2, 1, 1)
        self.sb_binary = QtWidgets.QSlider(self.gp_threshold)
        self.sb_binary.setMaximum(255)
        self.sb_binary.setProperty("value", 120)
        self.sb_binary.setOrientation(QtCore.Qt.Horizontal)
        self.sb_binary.setObjectName("sb_binary")
        self.gridLayout_2.addWidget(self.sb_binary, 4, 1, 1, 1)
        self.sb_binary_text = QtWidgets.QLabel(self.gp_threshold)
        self.sb_binary_text.setAlignment(QtCore.Qt.AlignCenter)
        self.sb_binary_text.setObjectName("sb_binary_text")
        self.gridLayout_2.addWidget(self.sb_binary_text, 4, 0, 1, 1)
        self.sb_ratio_text = QtWidgets.QLabel(self.gp_threshold)
        self.sb_ratio_text.setAlignment(QtCore.Qt.AlignCenter)
        self.sb_ratio_text.setObjectName("sb_ratio_text")
        self.gridLayout_2.addWidget(self.sb_ratio_text, 6, 0, 1, 1)
        self.LeftOperation.addWidget(self.gp_threshold)
        self.horizontalLayout.addWidget(self.widget, 0, QtCore.Qt.AlignTop)
        self.widget_2 = QtWidgets.QWidget(self.centralWidget)
        self.widget_2.setObjectName("widget_2")
        self.RightOperation = QtWidgets.QGridLayout(self.widget_2)
        self.RightOperation.setContentsMargins(11, 11, 11, 11)
        self.RightOperation.setSpacing(6)
        self.RightOperation.setObjectName("RightOperation")
        self.gp_show_image = QtWidgets.QGroupBox(self.widget_2)
        self.gp_show_image.setObjectName("gp_show_image")
        self.horizontalLayout_2 = QtWidgets.QHBoxLayout(self.gp_show_image)
        self.horizontalLayout_2.setContentsMargins(11, 11, 11, 11)
        self.horizontalLayout_2.setSpacing(6)
        self.horizontalLayout_2.setObjectName("horizontalLayout_2")
        self.lbl_image_origin = QtWidgets.QLabel(self.gp_show_image)
        self.lbl_image_origin.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.lbl_image_origin.setText("")
        self.lbl_image_origin.setAlignment(QtCore.Qt.AlignCenter)
        self.lbl_image_origin.setObjectName("lbl_image_origin")
        self.horizontalLayout_2.addWidget(self.lbl_image_origin)
        self.lbl_image_after = QtWidgets.QLabel(self.gp_show_image)
        self.lbl_image_after.setFrameShape(QtWidgets.QFrame.StyledPanel)
        self.lbl_image_after.setText("")
        self.lbl_image_after.setAlignment(QtCore.Qt.AlignCenter)
        self.lbl_image_after.setObjectName("lbl_image_after")
        self.horizontalLayout_2.addWidget(self.lbl_image_after)
        self.RightOperation.addWidget(self.gp_show_image, 0, 1, 1, 1)
        self.text_result = QtWidgets.QTextEdit(self.widget_2)
        self.text_result.setMaximumSize(QtCore.QSize(16777215, 100))
        self.text_result.setReadOnly(True)
        self.text_result.setObjectName("text_result")
        self.RightOperation.addWidget(self.text_result, 1, 1, 1, 1)
        self.gp_operate_image = QtWidgets.QGroupBox(self.widget_2)
        self.gp_operate_image.setObjectName("gp_operate_image")
        self.verticalLayout_4 = QtWidgets.QVBoxLayout(self.gp_operate_image)
        self.verticalLayout_4.setContentsMargins(11, 11, 11, 11)
        self.verticalLayout_4.setSpacing(6)
        self.verticalLayout_4.setObjectName("verticalLayout_4")
        self.btn_split = QtWidgets.QPushButton(self.gp_operate_image)
        self.btn_split.setObjectName("btn_split")
        self.verticalLayout_4.addWidget(self.btn_split)
        self.btn_gray = QtWidgets.QPushButton(self.gp_operate_image)
        self.btn_gray.setObjectName("btn_gray")
        self.verticalLayout_4.addWidget(self.btn_gray)
        self.btn_brinary = QtWidgets.QPushButton(self.gp_operate_image)
        self.btn_brinary.setObjectName("btn_brinary")
        self.verticalLayout_4.addWidget(self.btn_brinary)
        self.btn_location = QtWidgets.QPushButton(self.gp_operate_image)
        self.btn_location.setObjectName("btn_location")
        self.verticalLayout_4.addWidget(self.btn_location)
        self.btn_recognize = QtWidgets.QPushButton(self.gp_operate_image)
        self.btn_recognize.setObjectName("btn_recognize")
        self.verticalLayout_4.addWidget(self.btn_recognize)
        self.RightOperation.addWidget(self.gp_operate_image, 0, 2, 2, 1, QtCore.Qt.AlignTop)
        self.horizontalLayout.addWidget(self.widget_2)
        MainWindow.setCentralWidget(self.centralWidget)

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)
        MainWindow.setTabOrder(self.btn_openImg, self.ck_noline)
        MainWindow.setTabOrder(self.ck_noline, self.ck_print)
        MainWindow.setTabOrder(self.ck_print, self.ck_cut)
        MainWindow.setTabOrder(self.ck_cut, self.ck_join)
        MainWindow.setTabOrder(self.ck_join, self.sb_binary)
        MainWindow.setTabOrder(self.sb_binary, self.sb_ratio)
        MainWindow.setTabOrder(self.sb_ratio, self.btn_split)
        MainWindow.setTabOrder(self.btn_split, self.btn_gray)
        MainWindow.setTabOrder(self.btn_gray, self.btn_brinary)
        MainWindow.setTabOrder(self.btn_brinary, self.btn_location)
        MainWindow.setTabOrder(self.btn_location, self.btn_recognize)
        MainWindow.setTabOrder(self.btn_recognize, self.text_result)

        self.btn_openImg.clicked.connect(MainWindow.open_image)
        self.sb_binary.valueChanged.connect(MainWindow.sb_value_change)
        self.sb_ratio.valueChanged.connect(MainWindow.sb_value_change)
        self.ck_print.stateChanged.connect(MainWindow.ck_value_change)
        self.btn_split.clicked.connect(MainWindow.btn_click)
        self.btn_gray.clicked.connect(MainWindow.btn_click)
        self.btn_brinary.clicked.connect(MainWindow.btn_click)
        self.btn_location.clicked.connect(MainWindow.btn_click)
        self.btn_recognize.clicked.connect(MainWindow.btn_click)
        self.lbl_image_origin.installEventFilter(self)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        self.gp_file_option.setTitle(_translate("MainWindow", "文件选项"))
        self.btn_openImg.setText(_translate("MainWindow", "打开图片"))
        self.gp_ability_option.setTitle(_translate("MainWindow", "功能选项"))
        self.ck_cut.setText(_translate("MainWindow", "图像压缩"))
        self.ck_noline.setText(_translate("MainWindow", "去除杂线"))
        self.ck_print.setText(_translate("MainWindow", "印刷体"))
        self.ck_join.setText(_translate("MainWindow", "图像合并"))
        self.gp_threshold.setTitle(_translate("MainWindow", "阈值管理"))
        self.sb_ratio_value.setText(_translate("MainWindow", "0.30"))
        self.sb_binary_value.setText(_translate("MainWindow", "120"))
        self.sb_binary_text.setText(_translate("MainWindow", "二值阈值"))
        self.sb_ratio_text.setText(_translate("MainWindow", "字体斜率"))
        self.gp_show_image.setTitle(_translate("MainWindow", "图像预览区域"))
        self.text_result.setPlaceholderText(_translate("MainWindow", "识别结果将在此处显示"))
        self.gp_operate_image.setTitle(_translate("MainWindow", "图像操作"))
        self.btn_split.setText(_translate("MainWindow", "裁剪图片"))
        self.btn_gray.setText(_translate("MainWindow", "灰度预览"))
        self.btn_brinary.setText(_translate("MainWindow", "二值预览"))
        self.btn_location.setText(_translate("MainWindow", "位置预览"))
        self.btn_recognize.setText(_translate("MainWindow", "文字识别"))
